### ISSUE TITLE
Implement rebuild graph from item data mechanism.

### ISSUE THREAD

**Issue by jde** [coralproject/shelf#304]
Currently, graph relationships are only created/removed when items are upserted/deleted. There is no way to create the graph based on a new set of configuration.

We need a mechanism that can build the full graph from scratch based on an existing item set.

**Comment by jde** [comment]
Approach to consider: 
```
* empty cayley graph collections
* loop through all items and run infer relationships
```

**Comment by dwhitena** [comment]
@jde I don't think we have to necessary build from scratch if we say, update the predicate for one relationship.  Really this only requires an update for some of the graph relationships.  I think we should explore an approach similar to what we are doing with items, such that we can update the graph where it needs to be updated.

Of course a first step towards this may be a straight drop and rebuild of the graph.  So I will probably try that first.
